src/main.rs

```diff
@@ -60,6 +60,10 @@ impl State {
             self.hk = setup_hotkeys(self.settings.get_mut_inner_settings());
             self.settings.get_mut_inner_settings().rate =
                 self.voice.set_rate(self.settings.get_inner_settings().rate);
+            self.settings.get_mut_inner_settings().voice =
+                self.voice.set_voice_by_name(self.settings.get_mut_inner_settings().voice.clone());
+            self.settings.get_mut_inner_settings().available_voices =
+                self.voice.available_voice_names();
```

Comment:
Ah, no it doesn't. I think I had confused `SettingsWindow` with `Settings`. I've changed so that the voices list are passed from `main` straight to `SettingsWindow`.